### PR TITLE
add topic name metadata calling the guest handler

### LINKED ISSUE #25: Pass through message topic to guest
Currently only the message payload is passed through to the target component. It would be useful to additionally send the MQTT topic through. This would then allow the trigger to subscribe to wildcard topics, i.e. those containing `+` or `#` (which already works) but also provide some context to the component on the source of the message.

### PR DESCRIPTION
**Pull request by karthik2804**
The topic is now added as metadata that is passed to the guest.

closes #25

### CODE REVIEW

File: spin-mqtt.wit [hunk 1]
```diff
@@ -15,15 +15,20 @@ interface spin-mqtt-types {
       exactly-once,
   }
 
+  // metadata assocaited with the payload
```
Comment (fibonacci1729) [inline spin-mqtt.wit:18]:
```suggestion
  // metadata associated with the payload
```

**Review by fibonacci1729 — APPROVED** [review]

**Comment by suneetnangia** [comment]
Thanks much @karthik2804, will have a look tomorrow morning and merge.

**Review by suneetnangia — COMMENTED** [review]
Thanks @karthik2804, requested a small change in comment and if you have time, probably worth updating template?

**Comment by karthik2804** [comment]
updated the template and now it also has async on it.

File: src/lib.rs [hunk 2]
```diff
@@ -199,7 +206,7 @@ impl MqttTrigger {
                 Ok(Some(msg)) => {
                     // Handle the received message
                     _ = self
-                        .handle_mqtt_event(&component_id, msg.payload().to_vec())
+                        .handle_mqtt_event(&component_id, msg.payload().to_vec(), topic.clone())
```
[the remark this answers is not in the document]
Comment (suneetnangia) [reply src/lib.rs]: I see this is fixed as well. Closing it. Thanks.

**Review by suneetnangia — APPROVED** [review]

**Pull request merged**